internal/backend/message_tx.go

```diff
@@ -359,3 +359,30 @@ func txUpdateMessageID(ctx context.Context, tx *ent.Tx, oldID, newID string) err
 
 	return nil
 }
+
+func txMarkMessageAsDeleted(ctx context.Context, tx *ent.Tx, messageID string) error {
+	if _, err := tx.Message.Update().Where(message.MessageID(messageID)).SetDeleted(true).Save(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func txDeleteMessagesMarkedDeleted(ctx context.Context, tx *ent.Tx, messageIDs ...string) error {
+	if _, err := tx.Message.Delete().Where(message.And(message.MessageIDIn(messageIDs...), message.Deleted(true))).Exec(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
These are very long lines (go style usually limits col width to 120). Further, the `Where(...)` performs `And` by default, no need to explicitly call the `And` operator.

```suggestion
func txDeleteMessagesMarkedDeleted(ctx context.Context, tx *ent.Tx, messageIDs ...string) error {
	if _, err := tx.Message.Delete().Where(
		message.MessageIDIn(messageIDs...),
		message.Deleted(true),
	).Exec(ctx); err != nil {
		return err
	}

	return nil
}
```

Reply:
This function has been replaced with txDeleteMessages().